src/main/java/org/everit/osgi/dev/maven/statistic/GoogleAnalyticsTrackingServiceImpl.java

```diff
@@ -0,0 +1,170 @@
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.entity.UrlEncodedFormEntity;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.apache.http.message.BasicNameValuePair;
+
+/**
+ * Implementation of {@link GoogleAnalyticsTrackingService}.
+ */
+public class GoogleAnalyticsTrackingServiceImpl implements GoogleAnalyticsTrackingService {
+
+  /**
+   * Simple implementation of the {@link Runnable}. Responsible to send event the Google Analytics
+   * server.
+   */
+  private class EventSender implements Runnable {
+
+    private final String analyticsReferer;
+
+    private final String executedGoalName;
+
+    private final HttpClient httpClient;
+
+    private final String macAddressHash;
+
+    /**
+     * Constructor.
+     *
+     * @param analyticsReferer
+     *          the name of the referer that means who execute goal (example: eosgi-maven-plugin or
+     *          eclipse-e4-plugin).
+     * @param executedGoalName
+     *          the name of the executed goal.
+     * @param macAddressHash
+     *          the MAC address hash to anonym_user_id dimension.
+     */
+    EventSender(final String analyticsReferer, final String executedGoalName,
+        final String macAddressHash) {
+      this.analyticsReferer = analyticsReferer;
+      this.executedGoalName = executedGoalName;
+      this.macAddressHash = macAddressHash;
+      httpClient = new DefaultHttpClient();
+    }
+
+    @Override
+    public void run() {
+      HttpPost post = new HttpPost(GA_ENDPOINT);
+      List<NameValuePair> params = new ArrayList<>();
+      params.add(new BasicNameValuePair("v", "1")); // version
+      params.add(new BasicNameValuePair("tid", trackingId)); // tracking id
+      params.add(new BasicNameValuePair("cid", macAddressHash)); // client id
+      params.add(new BasicNameValuePair("t", "event")); // hit type
+      params.add(new BasicNameValuePair("ec", analyticsReferer)); // event category
+      params.add(new BasicNameValuePair("ea", executedGoalName)); // event action
+      params.add(new BasicNameValuePair("dimension1", macAddressHash)); // anonym_user_id
+      try {
+        UrlEncodedFormEntity entity =
+            new UrlEncodedFormEntity(params, "UTF-8");
+        post.setEntity(entity);
+        httpClient.execute(post);
+      } catch (IOException e) {
+        throw new RuntimeException(e);
+      }
+    }
+  }
+
+  private static final String GA_ENDPOINT =
+      "http://www.google-analytics.com/collect?payload_data";
+
+  private static final String UNKNOWN_MAC_ADDRESS = "UNKNOWN_MAC_ADDRESS";
+
+  private final long analyticsWaitingTimeInMs;
+
+  private final ConcurrentMap<Long, Thread> sendingEvents = new ConcurrentHashMap<>();
+
+  private final boolean skipTracking;
+
+  private final String trackingId = "UA-69304815-1";
```

Comment:
Initialize the trackingId in the constructor with a constructor argument. This argument should be the first.
The "private static final String TRACKING_ID = "UA-69304815-1";" constant can be defined and used in the AbstractEOSGiMojo where the GoogleAnalyticsTrackingServiceImpl is instantiated.